core/js/decanter.js

```diff
@@ -1,3 +1,434 @@
-/**
- * Empty until further notice.
- */
+// if NodeList doesn't support forEach, use Array's forEach()
+NodeList.prototype.forEach = NodeList.prototype.forEach || Array.prototype.forEach;
+
+//// TODO: move to components/main-nav.js file
+document.addEventListener( "DOMContentLoaded", event => {
+
+  const navClass = 'su-main-nav';
+
+  //////
+  // Keyboard helper functions
+  const isHome = theKey => ( theKey === 'Home' || theKey === 122 );
+  const isEnd = theKey => ( theKey === 'End' || theKey === 123 );
+  const isTab = theKey => ( theKey === 'Tab' || theKey === 9 );
+  const isEsc = theKey => ( theKey === 'Escape' || theKey === 'Esc' || theKey === 27 );
+  const isSpace = theKey => ( theKey === ' ' || theKey === 'Spacebar' || theKey === 32 );
+  const isEnter = theKey => ( theKey === 'Enter' || theKey === 13 );
+  const isLeftArrow = theKey => ( theKey === 'ArrowLeft' || theKey === 'Left' || theKey === 37 );
+  const isRightArrow = theKey => ( theKey === 'ArrowRight' || theKey === 'Right' || theKey === 39 );
+  const isUpArrow = theKey => ( theKey === 'ArrowUp' || theKey === 'Up' || theKey === 38 );
+  const isDownArrow = theKey => ( theKey === 'ArrowDown' || theKey === 'Down' || theKey === 40 );
+
+
+  //////
+  // Helper classes
+
+  /** Represent an item in a navigation menu */
+  class NavItem {
+    /**
+     * Create a NavItem
+     * @param {HTMLLIElement} item - the <li> that is the NavItem in the DOM
+     * @param {HTMLElement|Nav} nav - the Nav that contains the element. May be a main nav (<nav>) or a subnav (Nav)
+     */
+    constructor( item, nav ) {
+      this.item = item;
+      this.nav = nav;
+      this.link = this.item.querySelector( 'a' );
+      this.subNav = null;
+
+      this.item.addEventListener( 'keydown', this );
+
+      if ( this.isSubNavTrigger() ) {
+        this.subNav = new Nav( this );
+        theSubNavs.push( this ); // maintain globsl list of subnavs for closeAllSubNavs()
+        this.item.addEventListener( 'click', this );
+      }
+    }
+
+    isFirstItem() { return this.nav.items.indexOf( this ) === 0; }
+    isLastItem() { return this.nav.items.indexOf( this ) === this.nav.items.length - 1; }
+    isSubNavTrigger() { return this.item.lastElementChild.tagName.toUpperCase() === 'UL'; };
+    isSubNavItem() { return this.isSubNavTrigger() || this.nav.isSubNav(); }
+    isExpanded() { return this.link.getAttribute('aria-expanded') === 'true'; }
+    setExpanded( value ) { this.link.setAttribute( 'aria-expanded', value ); }
+
+    openSubNav() {
+      closeAllSubNavs();
+
+      if ( this.isSubNavTrigger() ) {
+        this.item.classList.add( 'su-main-nav__item--expanded' );
+        this.setExpanded( 'true' );
+        this.subNav.focusOn( 'first' );
```

Comment:
This is great. Thank you.

Reply:
Not sure exactly what you're referring to, but "great" works for me! :)